qemu/index.js

```diff
@@ -106,6 +106,75 @@ app.post('/machines', upload.array(), function (req, res) {
     });
 });
 
+app.get('/machines/status/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}', function (req, res) {
+  var machineId = req.path.substr(17);
+  let machineStatus = null;
+  return new Promise((resolve) => {
+    let socket = net.connect({path: drivePath + machineId + '.socket'}, () => {
+      return resolve(socket);
+    });
+  })
+    .then((socket) => {
+      return new Promise((resolve) => {
+        socket.write('info status\n', () => {
+          return resolve();
+        });
+      })
+        .then(() => {
+          return new Promise((resolve) => {
+            socket.on('data', (data) => {
+              var lines = data.toString().split('\n');
+              lines.forEach(function(line) {
+                if ( line.substr(0,9) === 'VM status' ) {
+                  machineStatus = line.split(':')[1].substr(1);
+                  return resolve();
+                }
+              });
+            });
+          });
+        })
+        .then(() => {
+          /**
+           * Qemu can return only 'running' or 'paused' but the socket a \r at the end
+           * of the string, so we need to know what is the string value to split it
+           * correctly.
+           */
+          machineStatus = machineStatus.substr(0, (machineStatus.substr(0, 1) === 'p') ? 6 : 7);
```

Comment:
Magic value again. This should be more explicit. Do not hesitate to provide output example in code comments.